pkg/services/ngalert/state/manager_test.go

```diff
@@ -1120,8 +1122,9 @@ func TestProcessEvalResults(t *testing.T) {
 						"alertname":                    "test_title",
 						"label":                        "test",
 					},
-					Values: make(map[string]float64),
-					State:  eval.NoData,
+					Values:      make(map[string]float64),
+					State:       eval.NoData,
+					StateReason: models.StateReasonNoData,
```

Comment:
Not sure I like this approach. The idea of `StateReason` is to contain the original State if it was "mapped". For example, if NoData is mapped to Alerting. then the state should be `Alerting (NoData)`. However, in this PR we seem to change that logic and set state reason always. This will affect state history which will show `NoData (NoData)` status.

Reply:
You are right, state history is being shown that way.

<img width="727" alt="Captura de pantalla 2023-02-22 a las 22 37 50" src="https://user-images.githubusercontent.com/2112640/220765126-428849c4-77d5-4a0a-b904-fb99e4227949.png">

Reply:
Correct, I would appreciate it if we can revert that piece. Thanks for the example.